packages/utils/src/composeAbortSignals.ts

```diff
@@ -10,29 +10,42 @@ export type ComposedAbortSignal = AbortSignal & { destroy: () => void }
  * aforementioned instance of AbortSignal will have more and more listeners added
  * but never cleaned.
  */
-export function composeAbortSignals(...signals: AbortSignal[]): ComposedAbortSignal {
-    if (signals.length === 0) {
-        throw new Error('must provide at least one AbortSignal')
-    }
+export function composeAbortSignals(
+    ...signals: (AbortSignal | undefined)[]
+): ComposedAbortSignal {
+    const abortController = new AbortController()
```

Comment:
(nitpick) This could be moved lower in the function as it is not needed e.g. if there is a pre-aborted signal.

Reply:
Neat! I'll adjust.